crates/oxc_linter/src/rules/unicorn/prefer_math_min_max.rs

```diff
@@ -0,0 +1,261 @@
+use crate::{context::LintContext, rule::Rule, utils::is_same_reference, AstNode};
+
+fn prefer_math_min_max_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(
+        "Prefer `Math.min()` and `Math.max()` over ternaries for simple comparisons.",
+    )
+    .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferMathMinMax;
+
+#[derive(Debug)]
+enum TypeOptions {
+    Min,
+    Max,
+    Unknown,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Prefers use of [`Math.min()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min)
+    /// and [`Math.max()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max) instead of
+    /// ternary expressions when performing simple comparisons for more concise, easier to understand, and less prone to errors.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```javascript
+    /// height > 50 ? 50 : height;
+    /// height > 50 ? height : 50;
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```javascript
+    /// Math.min(height, 50);
+    /// Math.max(height, 50);
+    /// ```
+    PreferMathMinMax,
+    pedantic,
+    fix
+);
+
+impl Rule for PreferMathMinMax {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::ConditionalExpression(conditional_expr) = node.kind() else {
+            return;
+        };
+
+        let Expression::BinaryExpression(test_expr) = &conditional_expr.test else {
+            return;
+        };
+
+        let condition_type =
+            is_min_max(test_expr, &conditional_expr.consequent, &conditional_expr.alternate, ctx);
+
+        if matches!(condition_type, TypeOptions::Unknown) {
+            return;
+        }
+
+        ctx.diagnostic_with_fix(prefer_math_min_max_diagnostic(conditional_expr.span), |fixer| {
+            let Some(consequent) = get_expr_value(&conditional_expr.consequent) else {
+                return fixer.noop();
+            };
+            let Some(alternate) = get_expr_value(&conditional_expr.alternate) else {
+                return fixer.noop();
+            };
+
+            return match condition_type {
+                TypeOptions::Max => fixer
+                    .replace(conditional_expr.span, format!("Math.max({consequent}, {alternate})")),
+                TypeOptions::Min => fixer
+                    .replace(conditional_expr.span, format!("Math.min({consequent}, {alternate})")),
+                TypeOptions::Unknown => fixer.noop(),
+            };
+        });
+    }
+}
+
+fn get_expr_value(expr: &Expression) -> Option<String> {
+    match expr {
+        Expression::NumericLiteral(lit) => Some(lit.to_string()),
+        Expression::UnaryExpression(lit) => {
+            let mut unary_str: Vec<Cow<str>> = Vec::new();
+
+            unary_str.push(Cow::Borrowed(lit.operator.as_str()));
```

Comment:
Since we're ultimately turning this into a `String`, I'm not sure if the `Cow` here is worth it: maybe we could just use a String directly?

```suggestion
            let mut unary_str: String = String::new();

            unary_str.push_str(lit.operator.as_str()));
```

I'd also recommend specifying some initial capacity too: `String::with_capacity(lit.operator.len())`, but it's probably not a big deal either way.

Reply:
oh yeah we should probs use cow in the diagnotic i think

Reply:
example>> 

https://github.com/oxc-project/oxc/blob/3d13c6d1f9dedb8ead75afa144ac2a7e0234614a/crates/oxc_linter/src/rules/eslint/func_names.rs#L430

https://github.com/oxc-project/oxc/blob/3d13c6d1f9dedb8ead75afa144ac2a7e0234614a/crates/oxc_linter/src/rules/eslint/func_names.rs#L446

Reply:
Rolled back `Cow` to `String` and added `Cow` in the diagnotic.